applications/StructuralMechanicsApplication/tests/cpp_tests/test_small_displacement_mixed_volumetric_strain_oss_element.cpp

```diff
@@ -0,0 +1,292 @@
+    p_node_2->FastGetSolutionStepValue(DISPLACEMENT) = aux_disp;
+    aux_disp[1] = 0.1;
+    p_node_3->FastGetSolutionStepValue(DISPLACEMENT) = aux_disp;
+
+    array_1d<double,3> aux_acc = ZeroVector(3);
+    p_node_1->FastGetSolutionStepValue(ACCELERATION) = aux_acc;
+    aux_acc[0] = 100.0;
+    p_node_2->FastGetSolutionStepValue(ACCELERATION) = aux_acc;
+    aux_acc[1] = 200.0;
+    p_node_3->FastGetSolutionStepValue(ACCELERATION) = aux_acc;
+
+    p_node_1->FastGetSolutionStepValue(VOLUMETRIC_STRAIN) = 0.01;
+    p_node_2->FastGetSolutionStepValue(VOLUMETRIC_STRAIN) = 0.01;
+    p_node_3->FastGetSolutionStepValue(VOLUMETRIC_STRAIN) = 0.02;
+
+    // Set fake projection values to calculate the residual
+    p_node_1->FastGetSolutionStepValue(DISPLACEMENT_PROJECTION) = aux_disp;
+    p_node_2->FastGetSolutionStepValue(DISPLACEMENT_PROJECTION) = aux_disp;
+    p_node_3->FastGetSolutionStepValue(DISPLACEMENT_PROJECTION) = aux_disp;
+    p_node_1->FastGetSolutionStepValue(VOLUMETRIC_STRAIN_PROJECTION) = 1e-3;
+    p_node_2->FastGetSolutionStepValue(VOLUMETRIC_STRAIN_PROJECTION) = 2e-3;
+    p_node_3->FastGetSolutionStepValue(VOLUMETRIC_STRAIN_PROJECTION) = 3e-3;
+
+    // Compute RHS, LHS and mass matrix
+    Vector RHS = ZeroVector(9);
+    Matrix LHS = ZeroMatrix(9,9);
+    Matrix MassMatrix = ZeroMatrix(9,9);
+
+    p_element->Initialize(r_process_info); // Initialize the element to initialize the constitutive law
+    p_element->FinalizeSolutionStep(r_process_info); // Fake call to the FinalizeSolutionStep in order to upgrade the subscales dynamic component
+    p_element->FinalizeSolutionStep(r_process_info); // Fake call to the FinalizeSolutionStep in order to upgrade the subscales dynamic component
+    p_element->CalculateLocalSystem(LHS, RHS, r_process_info);
+    p_element->CalculateMassMatrix(MassMatrix, r_process_info);
+
+    // Check RHS and LHS results
+    const double tolerance = 1.0e-5;
+    const std::vector<double> expected_RHS({51181.5777242, 51173.076373, -1831.8931645, -12703.0376604, -38461.5390116, 10570.705396, -38453.0376604, -12711.5390116, 3972.72623009});
+    const std::vector<double> expected_LHS_row_0({778846.153846, 9615.38461538, -327036.58062, -394230.769231, -384615.384615, -307578.803995, -384615.384615, 375000, -317307.692308});
+    const std::vector<double> expected_mass_row_0({83.3333333333, 0, 0, 41.6666666667, 0, 0, 41.6666666667, 0, 0});
+    const std::vector<double> expected_mass_row_2({0, 0, 0, 0, 0, 0, 0, 0, 0});
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(RHS, expected_RHS, tolerance)
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(row(LHS,0), expected_LHS_row_0, tolerance)
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(row(MassMatrix,0), expected_mass_row_0, tolerance)
+    KRATOS_EXPECT_VECTOR_RELATIVE_NEAR(row(MassMatrix,2), expected_mass_row_2, tolerance)
+}
+
+// KRATOS_TEST_CASE_IN_SUITE(SmallDisplacementMixedVolumetricStrainOssElementZienkiewiczPatch, KratosStructuralMechanicsFastSuite)
```

Comment:
¿?

Reply:
See the PR description. These will be activated in a follow up PR as require more developments.